cmake/tracy.cmake

```diff
@@ -1,17 +1,21 @@
 # Built as outlined in Tracy documentation (pg.12)
 set(TRACY_HOME ${PROJECT_SOURCE_DIR}/tt_metal/third_party/tracy)
 
+if(NOT ENABLE_TRACY)
```

Comment:
This is nice, this is letting the header itself to decide if it uses empty or real stubs right?

Reply:
That's correct.  The way it's working now:
We expose a flag in our repo that says "Tracy or No Tracy?"
Regardless of the flag, we ALWAYS:
* define a Tracy::TracyClient target
* Include headers from Tracy

**IF** we're going for a Yes Tracy route, then the Tracy::TracyTarget is defined by the Tracy project itself, with all the details a normal dependency would have.  But we tag on the required compiler flags, #define, etc.
**IF** we're going for a No Tracy route, then the Tracy::TracyTarget is actually just a shell that only provides access to the headers.  No library is build, or linked, no compiler flags or #defines are set.  

This is a little non-standard from typical dependencies, which generally you grab or not grab depending on a flag.  But Tracy is kind enough to provide stubs for all its types when in disabled mode.  So this way we get to use its stubs. Nice!
And with this encapsulated as a Target we depend on, we can project that onto any consumer of a module whenever we're using Tracy in a header file that's exposed to a consumer.  That will automatically provide the correct include dir path for the header to be resolved.  And it will also, conditionally, project the compiler flags and #define to have it all work the way it should.